test/grisp_connect_test_async.erl

```diff
@@ -0,0 +1,24 @@
+%% @doc Helper function for asynchronous function calls
+%% @end
+
+-module(grisp_connect_test_async).
+
+-export([async_eval/1]).
+-export([async_get_result/1]).
+
+%--- API -----------------------------------------------------------------------
+
+async_eval(Fun) ->
+    spawn_link(
+      fun() ->
+              Res = Fun(),
+              receive {'$async_get_result', Pid} ->
+                          Pid ! {'$async_result', Res} end
+      end).
+
+async_get_result(Pid) ->
+    unlink(Pid),
+    Pid ! {'$async_get_result', self()},
```

Comment:
Wee should still start monitoring the process after unlinking, to be sure we get a 'DOWN' if it dies while waiting...
And when demonitoring after getting the response we should use the option `[flush]` to not keep a monitoring message in the mailbox.

Reply:
I now monitor till the exit of the spawned process, because it should exit immediately after sending the result.

Reply:
If the process would crash just after unlinking, it would time out, so in production code we would need to monitor anyway, but for testing this is probably good enough.